pyproject.toml

```diff
@@ -14,27 +15,24 @@ readme = "README.md"
 keywords = ["netCDF", "UM", "postprocessing"]
 dynamic = ["version"]
 dependencies = [
-  "numpy",
-  "mule",
-  "cftime",
-  "netCDF4",
-  "scitools-iris",
-  "cf-units",
-  "PyYAML",
-  "f90nml",
+    "python >=3.10,<=3.12",
```

Comment:
Do we want to lock the max Python version to 3.12? I note that 3.13 was released about a week ago.

(And, in general, do we want to lock a max version of Python now, or leave it open, and only come back and set a max once we know there's a new Python that isn't compatible?)

Reply:
That's a good question. 
I am not sure what's the usual approach between:
- Constraining Python to an upper version and, whenever new Python versions get released, test them and, in case they are compatible, add the version to the upper constraint
- Only constraining for minimum Python versions and, whenever new Python versions get released, test them and, in case they are not compatible, add the version to the constraint

I don't have strong preferences here, but the 2nd approach might pose less burden on maintainance (providing we are lucky and newer versions of Python are compatible)

Reply:
On a second thought, when a new version is released we would still need to update the CI to use the new version as well (currently the matrix uses until py3.12).
Therefore, even from the maintenance burden standpoint, the two approaches are very similar.

Reply:
Might as well keep the upper constraint there, if for no other reason that to remind us/future devs where the testing has been completed to.

Reply:
Based on my comments above, my preference would then be the 1st approach (current version), to always ensure compatibility.